site/src/components/RuntimeErrorState/RuntimeErrorState.test.tsx

```diff
@@ -6,19 +6,21 @@ import {
   RuntimeErrorState,
 } from "./RuntimeErrorState"
 
-describe("RuntimeErrorState", () => {
-  beforeEach(() => {
-    // Given
-    const errorText = "broken!"
-    const errorStateProps = {
-      error: new Error(errorText),
-    }
-
-    // When
-    render(<RuntimeErrorState {...errorStateProps} />)
-  })
+const renderComponent = () => {
+  // Given
+  const errorText = "broken!"
+  const errorStateProps = {
+    error: new Error(errorText),
+  }
 
+  // When
+  return render(<RuntimeErrorState {...errorStateProps} />)
+}
+
+describe("RuntimeErrorState", () => {
   it("should show stack when encountering runtime error", () => {
+    renderComponent()
```

Comment:
What's the benefit of sticking this code scaffolding in the `renderComponent` util method instead of a `beforeEach`?

Reply:
Sure, it is a testing-library role: https://github.com/testing-library/eslint-plugin-testing-library/blob/main/docs/rules/no-render-in-setup.md I like not having too much nesting or things that are not running "close" on where it is running, but I have no strong opinions on that so we can just remove it if you think that is not usable at all

Reply:
Oh, interesting! Thanks for the link; I remember the nesting article you posted in the dev channel a while back, but I didn't realize the concern was not only abstraction but also mutable variables.
This looks good to me!